packages/preview-middleware-client/types/sap.ui.fl.d.ts

```diff
@@ -21,8 +21,17 @@ declare module 'sap/ui/fl/Change' {
         };
         fileName: string;
     }
+    export interface ChangeContentType {
+        fragmentPath: string;
+        index: number;
+        targetAggregation: string;
+        templateName?: string;
+    }
+
     interface Change {
         getDefinition: () => ChangeDefinition;
+        getContent: () => ChangeContentType;
+        setContent: (newContent: ChangeContentType) => void;
     }
```

Comment:
The content of a change can be different, the way you wrote it is each change has the same content structure/properties. 

The interface **ChangeContentType** only relates to the `addXML` change but not to the `codeExt` or `propertyChange`.

The interface could become generic, and accept a T type.

This is also valid for 

```ts
getContent: () => ChangeContentType;
setContent: (newContent: ChangeContentType) => void;
```

Reply:
Yes, you are right. I some how forgot about that. Refactored to generic types